src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -159,25 +160,33 @@ where
         let schedule_status = self
             .pick_and_assign_impl(compaction_group, request_channel)
             .await;
-        let cancel_task = match &schedule_status {
-            ScheduleStatus::Ok => None,
-            ScheduleStatus::NoTask | ScheduleStatus::PickFailure => None,
-            ScheduleStatus::NoAvailableCompactor(task)
-            | ScheduleStatus::AssignFailure(task)
-            | ScheduleStatus::SendFailure(task) => Some(task.clone()),
+        let (cancel_task, cancel_task_state) = match &schedule_status {
```

Comment:
nits: we can return a Option<(task, status)> instead of (Option<task>, Option<status>)

Reply:
Fix it